File: code_samples/haskell/data_structures_linked_list.hs
-- Data Structures: Linked List
-- AI/ML Training Sample

module LinkedList where

data LinkedList = LinkedList {
    getData :: String
} deriving (Show, Eq)

process :: LinkedList -> String -> LinkedList
process obj input = obj { getData = input }

validate :: LinkedList -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = LinkedList { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)

File: code_samples/scala/data_structures_linked_list.scala
/**
 * Data Structures: Linked List
 * AI/ML Training Sample
 */

class LinkedList {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object LinkedListExample {
  def main(args: Array[String]): Unit = {
    val instance = new LinkedList()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
